### PR TITLE
Add node watcher for node failure condition

### LINKED ISSUE #7: Add node watcher for node failure condition
**Is this a BUG REPORT or FEATURE REQUEST?**:

 /kind feature

**What happened**:
local pv monitor can not detect the node failure condition

**What you expected to happen**:
if node breaks down, local PVs on it can be marked

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

### PR DESCRIPTION
**Pull request by NickrenREN**
Add node watcher for node failure condition

**Which issue(s) this PR fixes** *(optional, close the issue(s) when PR gets merged)*:

Fixes #7

**Special notes for your reviewer**:

/cc @ddysher 

**Release note**:

```release-note
Add node watcher for node failure condition
```

### CODE REVIEW

**Comment by NickrenREN** [comment]
@ddysher  added deployment and README files in the last commit. PTAL, thanks

File: cmd/kube_storage_monitor/app/server.go [hunk 1]
```diff
@@ -14,12 +14,15 @@ import (
...
 type monitorOpt struct {
-	kube_storage_types []string
+	kube_storage_types  []string
+	enable_node_watcher bool
 }
 
 func (mo *monitorOpt) AddFlags(fs *pflag.FlagSet) {
 	fs.StringSliceVar(&mo.kube_storage_types, "kube-storage-types", mo.kube_storage_types, ""+
 		"kube-storage-types is the backend storage drivers type, such as: local_pv, cephfs, rbd... ")
+	fs.BoolVar(&mo.enable_node_watcher, "enable-node-watcher", mo.enable_node_watcher, ""+
```
Comment (ddysher) [inline cmd/kube_storage_monitor/app/server.go]: is node watcher only used for local PVs?

File: cmd/kube_storage_monitor/app/server.go [hunk 2]
```diff
@@ -38,25 +41,30 @@ func NewMonitorServerCommand() *cobra.Command {
...
-		os.Exit(1)
-	}
-	// run specific storage monitor
-	for _, sType := range mo.kube_storage_types {
-		switch sType {
-		// Add local_pv support at first
-		// If we get to support other storage types, need to add here
-		case "local_pv":
-			go local_pv.RunLocalPVMonitor()
+	if len(mo.kube_storage_types) > 0 {
```
Comment (ddysher) [inline cmd/kube_storage_monitor/app/server.go]: can I use both of the options? i.e. use `kube_storage_types` and `enable_node_watcher`

File: pkg/controller/node_failure_controller/node_failure_controller.go [hunk 3]
```diff
@@ -0,0 +1,406 @@
...
+		}
+	}
+
+	if markErrHappened {
+		return fmt.Errorf("error happened when marking local PVs")
+	}
+	return nil
+}
+
+func (watcher *NodeWatcher) markPV(pv *v1.PersistentVolume, ann, value string) error {
```
Comment (ddysher) [inline pkg/controller/node_failure_controller/node_failure_controller.go]: is this duplicate with markPV from local-pv-monitor?

File: build/kube_storage_monitor/node-watcher/README.md [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+Below is how to deploy node watcher to watch node failure conditions
+
+## Deployment
+
+### Create a kubernetes cluster containing two nodes(192.168.0.101, 192.168.0.105)
+``` kubectl get nodes
+NAME            STATUS     ROLES     AGE       VERSION
+192.168.0.101   Ready      <none>    8h        v1.10.0-alpha.1.5216+9b7439d77d0b29
+192.168.0.105   NotReady   <none>    30m       v1.10.0-alpha.1.1689+77d6ca8d4bb0b1-dirty
```
Comment (ddysher) [inline build/kube_storage_monitor/node-watcher/README.md]: This is already `NotReady`?

File: cmd/kube_storage_monitor/app/server.go (hunk 1 above)
Comment (NickrenREN) [reply cmd/kube_storage_monitor/app/server.go]: yes, networked storage PV will not be effected by node failure.

File: cmd/kube_storage_monitor/app/server.go (hunk 2 above)
Comment (NickrenREN) [reply cmd/kube_storage_monitor/app/server.go]: no, here node watcher is deployed separately

File: pkg/controller/node_failure_controller/node_failure_controller.go (hunk 3 above)
Comment (NickrenREN) [reply pkg/controller/node_failure_controller/node_failure_controller.go]: yes, most of them are the same, but we need to send record events and use client here, so i re-write it. We can revisit this later

File: build/kube_storage_monitor/node-watcher/README.md (hunk 4 above)
Comment (NickrenREN) [reply build/kube_storage_monitor/node-watcher/README.md]: Yes, we need to check that when node breaks down(NotReady for a period of time or Terminated), local PVs on that node are marked or not.

**Comment by NickrenREN** [comment]
I will update the top level README later, merging this ...

File: pkg/controller/node_failure_controller/node_failure_controller.go (hunk 3 above)
Comment (NickrenREN) [reply pkg/controller/node_failure_controller/node_failure_controller.go]: fix this in https://github.com/caicloud/kube-storage-monitor/pull/11

**Pull request merged**